### PR TITLE
[test] add cases for issues #690 #875 #876

### PR DESCRIPTION
**Pull request by barendgehrels**
This adds cases for issues #690 , #875 and #876 which I all closed today
Without rescaling they are all fine.

With rescaling and #690 there is something weird in the validity check, it might be related to the (robust) side_by_triangle determination too (see comment of @tinko92 in that issue). Therefore I changed the check_validity code in the test code too.

### CODE REVIEW

File: test/algorithms/set_operations/difference/difference.cpp [hunk 1]
```diff
@@ -627,7 +633,7 @@ int test_main(int, char* [])
     // Not yet fully tested for float and long double.
     // The difference algorithm can generate (additional) slivers
     // Many of the failures are self-intersection points.
-    BoostGeometryWriteExpectedFailures(12, 10, 17, 12);
+    BoostGeometryWriteExpectedFailures(19, 10, 17, 12);
```
Comment (barendgehrels) [inline test/algorithms/set_operations/difference/difference.cpp:636]: Side note: adding one case can indeed cause 7 extra reported issues (on number of geometries, area, validity, for a-b and b-a, s, etc)

File: test/algorithms/check_validity.hpp [hunk 2]
```diff
@@ -13,22 +13,38 @@
...
+                         Geometry const& geometry)
+{
+    std::string message;
+    bool const result = bg::is_valid(geometry, message);
+    if (! result)
+    {
+        std::cout << "WARNING: " << case_id << " Input [" << subcase
+                  << "] is not considered as valid ("
+                  << message << ") this can cause that output is invalid: "
+                  << case_id << std::endl;
```
Comment (awulkiew) [inline test/algorithms/check_validity.hpp:27]: You should probably use `BOOST_TEST_MESSAGE` here instead.

**Review by awulkiew — APPROVED** [review]

File: test/algorithms/set_operations/union/union.cpp [hunk 3]
```diff
@@ -430,6 +430,17 @@ void test_areal()
     TEST_UNION_REV(issue_566_a, 1, 0, -1, 214.3728);
     TEST_UNION_REV(issue_566_b, 1, 0, -1, 214.3728);
 
+#if ! defined(BOOST_GEOMETRY_USE_RESCALING) || defined(BOOST_GEOMETRY_TEST_FAILURES)
```
Comment (vissarion) [inline test/algorithms/set_operations/union/union.cpp:433]: I tested it with `side_robust` i.e. merging with https://github.com/boostorg/geometry/pull/873 and those tests are still failing, so probably not a side issue, and should be further investigated.

**Review by vissarion — APPROVED** [review]
Thanks, I am OK to merge these.

File: test/algorithms/check_validity.hpp (hunk 2 above)
Comment (barendgehrels) [reply test/algorithms/check_validity.hpp:27]: I don't know it yet, thanks.
I'll merge this (it's consistent to previous behavior) but will look at this in a later phase.

File: test/algorithms/set_operations/union/union.cpp (hunk 3 above)
Comment (barendgehrels) [reply test/algorithms/set_operations/union/union.cpp:433]: Good to know, thanks

**Pull request merged**